src/t8_forest/t8_forest_cxx.cxx

```diff
@@ -960,6 +989,218 @@ t8_forest_element_face_normal (t8_forest_t forest, t8_locidx_t ltreeid,
   }
 }
 
+int
```

Comment:
Shouldn't these functions be element-functions and therefore be implemented there?
This would avoid this very long switch blocks. 
On the other hand this would mean code-duplication for the bilinear-interpolated elements. If there was a way to call a more general function for these elements, this function would be much more pretty.

Reply:
No, as said above they are part of the geometry. I guess this handling will also change once we merge https://github.com/holke/t8code/tree/feature-geometry